test/fw/ptl/lib/ptl_resourceresv.py

```diff
@@ -652,7 +652,8 @@ def run(self):
             _p.sendline("exit")
             while True:
                 try:
-                    _p.read_nonblocking(timeout=5)
+                    # When -1 (default), use self.timeout
+                    _p.read_nonblocking()
```

Comment:
LGTM, @RKORavi  it would be great if you can simulate a time out during non blocking read

Reply:
while creating "_p"  value of timeout is set by "pexpect_timeout = 15". So self.timeout in this case is 15 sec. does it sufficient for interactive job?

Reply:
Hey @birbalsain, timeout in this case will be 30 secs. Because when we do not pass timeout it will use self.timeout value in pexpect. Please refer this link: https://pexpect.readthedocs.io/en/3.x/_modules/pexpect.html
I will test it again with job's sleep greater than 30 secs.

Reply:
@birbalsain yes, you are right. timeout will be 15 & not 30 secs. Thanks for pointing that out. I will make the changes accordingly.